exercises/largest-series-product/largest_series_product.cr

```diff
@@ -0,0 +1,27 @@
+class Series
+  def initialize(series : String)
+    if series =~ /\D/
+      @series = [] of Int64
+    else
+      @series = series.chars.map(&.to_i).map(&.to_i64)
+    end
+  end
+
+  def largest_product(span : Int32) : Int64
+    return -1i64 if span > @series.size || (@series.empty? && span > 0) || span < 0
+    return 1i64 if span == 0 
+
+    product_from_sequence(create_sequences(span))
+  end
+
+  private def create_sequences(span : Int32) : Array(Array(Int64))
+    seq = @series.map_with_index do |n, i|
+      @series[i, span]
+    end
+    seq.select{|s|s.size == span}
```

Comment:
Hmmm. I've always thought that looked a bit strange. `{|s| s.size == span}` looks better to me because it provides a clear advantage over `do |s| s.size == span end`. I don't really see why it should be different than something else using {} like a hash. Probably just stuck in my ways after 10 years in ruby. I'll look into the logic behind that a bit more. Thanks for linking that to me.